lib/bamboo/adapters/smtp_adapter.ex

```diff
@@ -356,9 +356,15 @@ defmodule Bamboo.SMTPAdapter do
     |> raise_on_missing_configuration(config)
   end
 
-  defp format_email({nil, email}, _format), do: email
-  defp format_email({name, email}, true), do: "#{rfc822_encode(name)} <#{email}>"
-  defp format_email({_name, email}, false), do: email
+  defp puny_encode(email) do
+    [local_part, domain_part] = String.split(email, "@")
+    Enum.join([local_part, :idna.utf8_to_ascii(domain_part)], "@")
```

Comment:
```suggestion
    local_part <> "@" <> :idna.utf8_to_ascii(domain_part)
```

Reply:
@babariviere I realized this will not work because the result of `:idna.utf8_to_ascii` is actually a charlist, not a string.